src/NzbDrone.Common/Http/Dispatchers/ManagedHttpDispatcher.cs

```diff
@@ -270,10 +283,9 @@ private static async ValueTask<Stream> onConnect(SocketsHttpConnectionContext co
                 }
                 catch
                 {
-                    // very naively fallback to ipv4 permanently for this execution based on the response of the first connection attempt.
-                    // note that this may cause users to eventually get switched to ipv4 (on a random failure when they are switching networks, for instance)
-                    // but in the interest of keeping this implementation simple, this is acceptable.
-                    useIPv6 = false;
+                    // fallback to ipv4 permanently if ipv6 connection has failed and a routable ipv4 is available.
+                    // if has_routable_ipv4_address not use_ipv6
+                    useIPv6 = !HasRoutableIPv4Address();
```

Comment:
```suggestion
                    // Do not retry IPv6 if a routable IPv4 address is available, otherwise continue to attempt IPv6 connections.
                    useIPv6 = !HasRoutableIPv4Address();
```